stl/inc/chrono

```diff
@@ -5486,34 +5497,79 @@ struct formatter<_CHRONO day, _CharT> {
         }
 
         for (const auto& _Spec : _List) {
-            if (_Spec._Type != '\0' && _Spec._Type != 'd' && _Spec._Type != 'e') {
-                _THROW(format_error("Invalid type for type day."));
-            }
-            if (_Spec._Modifier != '\0' && _Spec._Modifier != 'O') {
-                _THROW(format_error("Invalid modifier for type day."));
+            if (_Spec._Type != '\0' && _RANGES find(_Legal_types, _Spec._Type) != _Legal_types.end()) {
+                _THROW(format_error("Invalid type."));
             }
+            _Check_modifier(_Spec._Type, _Spec._Modifier);
         }
 
         return _Res_iter;
     }
 
-    template <class _FormatContext>
-    typename _FormatContext::iterator format(const _CHRONO day& _Val, _FormatContext& _FormatCtx) {
-        using _Ctx_iter = typename _FormatContext::iterator;
+    void _Check_modifier(const _CharT _Type, const _CharT _Modifier) {
+        if (_Modifier == _CharT{'\0'}) {
+            return;
+        }
+
+        const bool _O_mod = _Modifier == _CharT{'O'};
+        const bool _E_mod = _Modifier == _CharT{'E'};
+        (void) _E_mod;
 
+        switch (static_cast<char>(_Type)) {

```

Comment:
This is a lot of control flow. Can we replace this with a static table? From a quick scan of the Standard table, at most one of O and E are permitted for each type, so we could simply have an array of pairs (with `'\0'` if no modifier is permitted at all). If multiple modifiers are permitted, we could instead use flags. Then the control flow would be a simple scan. (We could optimize it later if it turns out to be a bottleneck.)

Reply:
Used an enum instead of enum class because bitwise ops on enum classes are yucky. Since the scope is so small, I think it's fine.